crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -691,38 +653,27 @@ impl<'w, T: Component> Fetch<'w> for ReadFetch<'w, T> {
         );
     }
 
-    #[inline]
-    unsafe fn archetype_fetch(&mut self, archetype_index: usize) -> Self::Item {
+    #[inline(always)]
+    unsafe fn fetch(&mut self, entity: &Entity, table_row: &usize) -> Self::Item {
         match T::Storage::STORAGE_TYPE {
             StorageType::Table => {
-                let (entity_table_rows, table_components) = self
-                    .entity_table_rows
-                    .zip(self.table_components)
+                let components = self
+                    .table_components
                     .unwrap_or_else(|| debug_checked_unreachable());
-                let table_row = *entity_table_rows.get(archetype_index);
-                table_components.get(table_row).deref()
+                components.get(*table_row).deref()
             }
             StorageType::SparseSet => {
-                let (entities, sparse_set) = self
+                let (_, sparse_set) = self
```

Comment:
This feels like it should no longer use a zip?

Reply:
This indeed is an artifact of this change. Not sure how the ptr zip function works here.